### PR TITLE
recreate the hub kubeconfig when addon deploy option changes

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>

### CODE REVIEW

**Comment by zhujian7** [comment]
cc @elgnay @qiujian16

**Comment by elgnay** [comment]
nit: It might be better to include `installationNamespace` and `addOnAgentRunningOutsideManagedCluster` when calculating the hash of the registrationConfig. Then we can keep the logic of the registration controller neat.

**Comment by qiujian16** [comment]
yes, I agree with @elgnay putting this into hash is better.

File: pkg/spoke/addon/configuration.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ type registrationConfig struct {
 	addOnAgentRunningOutsideManagedCluster bool
 }
 
+type addonInstallOption struct {
+	InstallationNamespace             string
+	AgentRunningOutsideManagedCluster bool
```
Comment (qiujian16) [inline pkg/spoke/addon/configuration.go]: should these have json marker? also in registrationConfig, we should put it in the separate struct so we do not duplicate it, right?

Comment (zhujian7) [reply pkg/spoke/addon/configuration.go]: Do you mean we put addonInstallOption inline in registrationConfig, right？

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**